api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,122 @@
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final LeonardoMapper leonardoMapper;
+  private final Provider<DbUser> userProvider;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceAuthService workspaceAuthService,
+      WorkspaceDao workspaceDao,
+      Provider<DbUser> userProvider,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.userProvider = userProvider;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String firecloudWorkspaceName = dbWorkspace.getFirecloudName();
+    String googleProject = dbWorkspace.getGoogleProject();
+    workspaceAuthService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
+    workspaceAuthService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
+    try {
+      LeonardoGetPersistentDiskResponse response;
+      String pdNamePrefix = userProvider.get().getUserPDNamePrefix();
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false)
+              .stream()
+              .filter(r -> r.getName().startsWith(pdNamePrefix))
+              .sorted((r1,r2) -> -1 * r1.getId().compareTo(r2.getId()))
+              .collect(Collectors.toList());
+      if (!responseList.isEmpty()){
+        response = leonardoNotebooksClient.getPersistentDisk(
+            googleProject, responseList.get(0).getName());
+        if (LeonardoDiskStatus.FAILED.equals(response.getStatus())) {
+          log.warning(
+              String.format(
+                  "Observed Leonardo persistent disk with unexpected failed status:\n%s",
+                  response.getStatus()));
+        }
+        return ResponseEntity.ok(leonardoMapper.toApiDisk(response));
+      } else {
+        return ResponseEntity.notFound().build();
+      }
+    } catch (NotFoundException e) {
+      return ResponseEntity.notFound().build();
+    }
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> deleteDisk(String workspaceNamespace) {
```

Comment:
Hmm. Should this just take `diskName`? Shouldn't the client know this information? This would make things a bit more explicit, so there are fewer cases where you might accidentally delete the wrong disk.